webapp/channels/src/components/channel_view/channel_view.tsx

```diff
@@ -159,6 +159,31 @@ export default class ChannelView extends React.PureComponent<Props, State> {
                     <AdvancedCreatePost/>
                 </div>
             );
+        } else {
+            createPost = (
+                <div
+                    className='post-create__container'
+                    id='post-create'
+                >
+                    <div
+                        className='channel-archived__message'
+                    >
+                        <FormattedMarkdownMessage
+                            id='noRolesMessage'
+                            defaultMessage='Roles for this channel are not yet loaded. Please wait or reload the app.'
+                        />
+                        <button
+                            className='btn btn-primary channel-archived__close-btn'
+                            onClick={this.onClickCloseChannel}
+                        >
+                            <FormattedMessage
+                                id='center_panel.archived.closeChannel'
+                                defaultMessage='Close Channel'
```

Comment:
Regardless of how we handle my last comment, I don't think we should show the user this button though. In the other cases, we give the user a way out of the channel because they can't interact with it at all, but in this case, they should be able to interact with the channel in a few seconds once everything is loaded